x-pack/test/api_integration/apis/siem/saved_objects/notes.ts

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import expect from '@kbn/expect';
+import gql from 'graphql-tag';
+
+import { allTimelinesQuery } from '../../../../../legacy/plugins/siem/public/containers/timeline/all/index.gql_query';
+import { persistTimelineNoteMutation } from '../../../../../legacy/plugins/siem/public/containers/timeline/notes/persist.gql_query';
+import { GetAllTimeline } from '../../../../../legacy/plugins/siem/public/graphql/types';
+
+import { KbnTestProvider } from '../types';
+
+const notesPersistenceTests: KbnTestProvider = ({ getService }) => {
+  const esArchiver = getService('esArchiver');
+  const client = getService('siemGraphQLClient');
+
+  describe('Note - Saved Objects', () => {
+    beforeEach(() => esArchiver.load('empty_kibana'));
+    afterEach(() => esArchiver.unload('empty_kibana'));
+
+    describe('create a note', () => {
+      it('should return a timelineId, timelineVersion, noteId and version', async () => {
+        const myNote = 'world test';
+        const response = await client.mutate<any>({
+          mutation: persistTimelineNoteMutation,
+          variables: {
+            noteId: null,
+            version: null,
+            note: { note: myNote, timelineId: null },
+          },
+        });
+        const { note, noteId, timelineId, timelineVersion, version } =
+          response.data && response.data.persistNote.note;
+
+        expect(note).to.be(myNote);
+        expect(noteId).to.not.be.empty();
+        expect(timelineId).to.not.be.empty();
+        expect(timelineVersion).to.not.be.empty();
+        expect(version).to.not.be.empty();
+      });
+
+      it('if noteId exist return existing timelineId, timelineVersion noteId and version', async () => {
+        const responseData = await client.query<GetAllTimeline.Query>({
+          query: allTimelinesQuery,
+          variables: {
+            search: '',
+            pageInfo: { pageIndex: 1, pageSize: 10 },
+            sort: { sortField: 'updated', sortOrder: 'desc' },
+          },
+        });
+        if (
+          responseData.data.getAllTimeline.timeline &&
+          responseData.data.getAllTimeline.timeline.length > 0 &&
+          responseData.data.getAllTimeline.timeline[0] != null &&
+          responseData.data.getAllTimeline.timeline[0]!.notes &&
+          responseData.data.getAllTimeline.timeline[0]!.notes.length > 0 &&
+          responseData.data.getAllTimeline.timeline[0]!.notes![0] != null
```

Comment:
If any one of the conditions above are not `true`, will this test run to completion without executing any `expect` statements? Could this `if` check be eliminated with the use of a `lodash/fp` `get`?

Reply:
there is no lodash library injected in the api_integration

Reply:
Hey, that's a good catch here!!! It was too easy, I am fixing it